backend/app/controllers/reading_lists_controller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class ReadingListsController < ApplicationController
+  before_action :authorize_request
+
+  def create
+    if reading_list.valid?
+      render(json: { data: reading_list }, status: :created)
+    else
+      render(json: { errors: reading_list.errors.full_messages }, status: :unprocessable_entity)
+    end
+  end
+
+  def update
+    if reading_list.blank?
+      render(json: { errors: ["Reading list not found"] }, status: :not_found)
+    elsif reading_list.update(reading_list_params)
+      render(json: { data: reading_list }, status: :ok)
+    else
+      render(json: { errors: reading_list.errors.full_messages }, status: :unprocessable_entity)
+    end
+  end
+
+  def destroy
+    if reading_list.present?
+      reading_list.destroy
+      render(json: { data: { message: "Successfully deleted" } }, status: :ok)
+    else
+      render(json: { errors: ["Reading list not found"] }, status: :not_found)
+    end
+  end
+
+  private
+
+  def reading_list_params
+    params.require(:reading_list).permit(:book_id, :status).merge({ user_id: @user.id })
+  end
+
+  def reading_list
+    @reading_list ||= if params[:id].present?
+      @user.reading_lists.find_by(id: params[:id])
+    else
+      ReadingList.create(reading_list_params)
+    end
```

Comment:
Let's change this to properly convey the method name. 

The method name says `reading_list` but there's a hidden mutation in the method. 

what I'd recommend is for the `reading_list` method to only return a reading list. Move the create reading_list in the create method. 

```suggestion
    @reading_list ||= @user.reading_lists.find_by(id: params[:id])
```